core/hoverfly.go

```diff
@@ -307,6 +307,20 @@ func (hf *Hoverfly) Save(request *models.RequestDetails, response *models.Respon
 		}
 	}
 
+	var headers map[string][]string
+
+	if headersToSave != nil && len(headersToSave) == 0 {
```

Comment:
I don't actually follow this. It looks like you are saving all headers whether it's an empty slice or a `nil` slice. I would see:

- Nil list means there is no whitelist, so save everything
- Empty list means there is a whitelist, containing no entries, so save nothing
- List with elements means these headers are whitelisted, so save them only